Assets/Scripts/Managers/InventoryManager.cs

```diff
@@ -19,33 +20,56 @@ public class InventoryManager : Singleton<InventoryManager>
         public IItemHandler ItemHandler => playerInventory;
 
         /// <summary>
-        /// The currently equipped item
+        /// The damage boost of the currently equipped weapon. Returns 0 if no item is equipped.
         /// </summary>
-        public Equipable CurrentlyEquippedItem { get; private set; }
+        public int DamageBoostFromEquipable => CurrentlyEquippedItem ? CurrentlyEquippedItem.DamageBoost : 0;
 
         /// <summary>
-        /// The damage boost of the currently equipped weapon. Returns 0 if no item is equipped.
+        /// Whether the inventory still has empty slots. Used for determining if an item can be added.
         /// </summary>
-        public int DamageBoostFromEquipable => CurrentlyEquippedItem ? CurrentlyEquippedItem.DamageBoost : 0;
+        public bool HasEmptySlots { get; set; }
+
+        /// <summary>
+        /// All items that still have an not-full stack. Used for determining if an item can be added.
+        /// </summary>
+        public HashSet<Item> NotFullItemStacks { get; set; }
 
         /// <summary>
         /// The currently equipped ItemButton (Visually equipped item)
         /// </summary>
         private ItemButton _currentlyEquippedItemButton;
 
+        /// <summary>
+        /// The currently equipped item
+        /// </summary>
+        private Equipable CurrentlyEquippedItem { get; set; }
+
         /// <summary>
         /// Adds an item to the player's inventory.
         /// </summary>
         /// <param name="item">The item to be added</param>
         /// <param name="amount">The amount</param>
-        public void AddItem(Item item, int amount = 1) => playerInventory.AddItem(item, amount);
+        public bool AddItem(Item item, int amount = 1)
+        {
+            for (int i = 0; i < amount; i++)
+            {
+                if (!HasEmptySlots && !NotFullItemStacks.Contains(item)) return false;
+                playerInventory.AddItem(item);
```

Comment:
May add some items and still return false if not all items could be added. Is this intended/do you have that in mind when calling the method? Might wanna add some info to the doc here :)

Reply:
Oh damn, you're right